test/test_unary_ufuncs.py

```diff
@@ -137,6 +132,37 @@ def generate_numeric_tensors(device, dtype, *,
 
     return chain(empty_tensors, scalar_tensors, small_tensors, (medium_tensor,), (large_tensor,))
 
+
+def generate_numeric_tensors_hard(device, dtype):
```

Comment:
Add a comment here explaining the function and how it's intended to be used. Let's clamp this one to the domain, too.

Reply:
Added in the front